### DISCUSSION TITLE
Why does poetry remove recalculate dependencies?

### DISCUSSION THREAD

**Discussion by sndrtj** [python-poetry/poetry#7238] [Q&A] [answered]
I've found that `poetry remove` causes all dependencies to be recalculated. I'm unfortunately working with some repos where that step can take a very long time, making otherwise seemingly simple removals take quite a bit.  Ought it not to be enough to only consider the package to be removed, and any potential dependencies of that package that aren't used by any other package(s) in the lock file?  

I suppose there can be cases where a package removal loosens constraints on other dependencies. However, I don't expect `poetry remove` to actually update these packages unless I specifically ask it to. 

So why does poetry remove recalculate all dependencies? There's probably a good reason, but at current it's beyond me.

**Comment by radoering** [comment, marked as answer]
`poetry add` and `poetry remove` are quite similar at the moment. Both are similar to editing pyproject.toml and running `poetry lock --no-update`. That's quite simple to implement and maintain.

I've never thought about it but it's probably possible to implement `poetry remove` more performant. However, I'm not sure it's worth it because in my experience most people don't care much about performance of `poetry remove`. Further, there are probably some pitfalls on the way (e.g. extras). That is not to say, I'm totally opposed, but it's a trade-off between performance and maintainability that has to be considered carefully.

**Reply by sndrtj** [reply]
That's fair enough. From my own experience, `poetry remove` indeed is a rather infrequently used command.
